### PR TITLE
Draft creation for nodes 

### PR DESCRIPTION
**Pull request by Gauravano**
Fixes part 1 of #2307 

Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

File: .rubocop-http---shopify-github-io-ruby-style-guide-rubocop-yml [hunk 1]
```diff
@@ -1080,9 +1080,6 @@ Performance/FixedSize:
 Performance/FlatMap:
   EnabledForFlattenWithoutParams: false
 
-Performance/HashEachMethods:
```
Comment (jywarren) [inline .rubocop-http---shopify-github-io-ruby-style-guide-rubocop-yml]: Hmm, what does this removal do?

Comment (Gauravano) [reply .rubocop-http---shopify-github-io-ruby-style-guide-rubocop-yml]: Actually, I don't know Jeff, every time I change code this file changes automatically. Can you specify the purpose of this file? Thanks.

**Comment by Gauravano** [comment]
> I am wondering if we need to add a test that shows that you can't view a status == 3 node unless you're the author?

This test would be added when I modify views as per the status.

File: .rubocop-http---shopify-github-io-ruby-style-guide-rubocop-yml (hunk 1 above)
Comment (jywarren) [reply .rubocop-http---shopify-github-io-ruby-style-guide-rubocop-yml]: Ah -- @ryzokuken @siaw23 do you know why?

**Comment by jywarren** [comment]
> This test would be added when I will modify views as per the status.

But we can do unit and functional tests even without a view, if you have changed a controller. You don't have to, of course! But it could be nice to just complete the back-end functionality, test and merge it, then add UI. Modularity! 

Great work on this! 👍

**Comment by Gauravano** [comment]
Ok writing tests

**Comment by Gauravano** [comment]
ok @jywarren  tests are added for 1st part

File: config/locales/controllers/notes_controller/en.yml [hunk 2]
```diff
@@ -16,3 +16,4 @@ en:
     highly_liked_research_notes: "Highly liked research notes"
     popular_research_notes: "Popular research notes"
     more_than_one_contributor: "You cannot delete a wiki page once someone else has begun to contribute to it"
+    saved_as_draft: "Your contribution has been saved as draft and will be visisble to you only"
```
Comment (seafr) [inline config/locales/controllers/notes_controller/en.yml]: A very tiny change needed with the word `visible` here.

Comment (Gauravano) [reply config/locales/controllers/notes_controller/en.yml]: thanks @seafr and changes done

**Comment by jywarren** [comment]
This looks great -- is it ready to merge, then? Awesome!!

**Pull request merged**